packages/migrator/src/index.ts

```diff
@@ -403,3 +403,57 @@ class RepairAction extends CommandLineAction {
 export interface RepairOptions {
   dryRun?: boolean
 }
+
+type LogMessage = Record<string, unknown>
+
+/**
+ * Logs messages to console. Known events are prettified to strings, unknown
+ * events or unexpected message properties in known events are logged as objects.
+ */
+export const prettyLogger: NonNullable<SlonikMigratorOptions['logger']> = {
+  info: message => prettifyAndLog('info', message),
+  warn: message => prettifyAndLog('warn', message),
+  error: message => prettifyAndLog('error', message),
+  debug: message => prettifyAndLog('debug', message),
+}
+
+function prettifyAndLog(level: keyof typeof prettyLogger, message: LogMessage) {
+  const MESSAGE_FORMATS: Record<string, (msg: LogMessage) => [string, LogMessage]> = {
```

Comment:
Typing nit: this is internal so not all that big a deal, but this typedef would allow usage like `MESSAGE_FORMATS.foobar123(...)` which will fail at runtime. You could make an identity function with a typearg to get the restriction you want without losing specificity:

```ts
  const createMessageFormats = <T extends Record<string, (msg: LogMessage) => [string, LogMessage]>>(formats: T) => formats;
  const MESSAGE_FORMATS = createMessageFormats({...})
```

There's a possibility will have a less awkward way to do this some day: https://github.com/microsoft/TypeScript/issues/47920

Reply:
Identity function used now.

Reply:
Soo.... incredibly hot off the press, TypeScript team have now fixed that issue! https://github.com/microsoft/TypeScript/pull/46827 - probably will take a little time to be released, but if it comes out before I have a chance to merge this, @samuliasmala want to be the first in the wild user of the `satisfies` operator??